fabric-transfer-api-v1/src/main/java/net/fabricmc/fabric/impl/transfer/TransferApiImpl.java

```diff
@@ -89,4 +88,54 @@ public T next() {
 			}
 		};
 	}
+
+	public static <T> Iterator<StorageView<T>> filterEmptyViews(Iterator<StorageView<T>> iterator) {
+		return new Iterator<>() {
+			StorageView<T> next;
+
+			{
+				findNext();
+			}
+
+			private void findNext() {
+				while (iterator.hasNext()) {
+					next = iterator.next();
+
+					if (next.getAmount() > 0 && !next.isResourceBlank()) {
+						return;
+					}
+				}
+
+				next = null;
+			}
+
+			@Override
+			public boolean hasNext() {
+				return next != null;
+			}
+
+			@Override
+			public StorageView<T> next() {
+				if (!hasNext()) {
+					throw new NoSuchElementException();
+				}
+
+				return next;
+			}
```

Comment:
This never calls `findNext()`, so next is never null or advanced, causing an infinite loop.